airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -199,3 +203,6 @@ class TIRunContext(BaseModel):
 
     connections: Annotated[list[ConnectionResponse], Field(default_factory=list)]
     """Connections that can be accessed by the task instance."""
+
+    start_date: Annotated[UtcDateTime, Field(title="Start Date")]
+    """Start date of the task instance."""
```

Comment:
This info is readily available here:

https://github.com/apache/airflow/blob/41b151e7dde473ec445f9f78fb4b8db826c368fc/task_sdk/src/airflow/sdk/execution_time/supervisor.py#L599-L605

I am wondering what would be the best place to hook the `on_start` listener.

Reply:
Yeah - this is on supervisor though, the start_date is not passed to the task runner. I can move it into `StartupDetails` though rather than roundtrip through API.

Reply:
Put it into `StartupDetails`.